plugins/module_utils/foreman_helper.py

```diff
@@ -775,7 +775,8 @@ def find_cluster(self, name, compute_resource):
 
         # workaround for https://projects.theforeman.org/issues/31874
         if compute_resource['provider'].lower() == 'vmware':
-            cluster['_api_identifier'] = cluster['name']
+            path_or_name = cluster.get('full_path', cluster['name'])
+            cluster['_api_identifier'] = six.moves.urllib.parse.quote(path_or_name, safe='')
```

Comment:
the double quoting (here and in apypie) is required for https://projects.theforeman.org/issues/35438